script.js

```diff
@@ -0,0 +1,448 @@
+            elem.classList.toggle('keyboard__keys_active');
+            toDeactiveByClick(elem);
+            toDeactiveByPress(elem);
+            break;
+         case 'Shift':
+            toActiveCapsOrShift(elem);
+            changeFirstRow();
+            toDeactiveByClick(elem);
+            toDeactiveByPress(elem);
+            break;
+         case 'Backspace':
+            toBackspace();
+            break;
+         case 'CapsLock':
+            toActiveCapsOrShift(elem);
+            break;
+         case 'Delete':
+            toDelete(input, cursorPos);
+            break;
+         default:
+            if (document.querySelectorAll('.keyboard__keys_upperCase').length === 0) {
+               input.innerHTML += elem.innerHTML;
+            }
+            else {
+               input.innerHTML += elem.innerHTML.toUpperCase();
+            }
+      }
+      toActive(elem);
+   });
+
+   elem.addEventListener('mouseup', function () {
+      if (elem == shiftLeft || elem == shiftRight || elem == capsLk || elem == ctrlLeft) {
+         return;
+      }
+      else {
+         elem.classList.remove('keyboard__keys_active');
+         if (ctrlLeft.classList.contains('keyboard__keys_active') && elem == altLeft) {
+            elem.classList.add('keyboard__keys_active');
+            checkActiveKeys(ctrlLeft, altLeft);
+            keyboardKeys.forEach(key =>
+               key.classList.remove('keyboard__keys_active'));
+         }
+      }
+   })
+});
+
+/* нажатие на кнопку физической клавиатуры */
+
+document.addEventListener('keypress', function (event) {
+
+})
+
+document.addEventListener('keydown', function (event) {
+
+   /* подсветка нужной клавиши */
+
+   if (event.which == 16) {
+      let shift = (event.location == 1) ? shiftLeft : shiftRight;
+      shift.classList.add('keyboard__keys_active');
+      keyboardKeysRegular.forEach(elem => {
+         elem.classList.add('keyboard__keys_upperCase');
+      });
+      changeFirstRow();
+      return;
+   }
+   if (event.which == 18) {
+      let alt = (event.location == 1) ? altLeft : altRight;
+      toActive(alt);
+      event.preventDefault();
+      checkActiveKeys(ctrlLeft, altLeft);
+      return;
+   }
+   if (event.which == 17) {
+      let ctrl = (event.location == 1) ? ctrlLeft : ctrlRight;
+      ctrl.classList.add('keyboard__keys_active');
+      event.preventDefault();
+      checkActiveKeys(ctrlLeft, altLeft);
+      return;
+   }
+   toActive(document.querySelector('.keyboard__keys[data="' + event.which + '"]'));
+
+   /* вывод при нажатии клавиши */
+
+   switch (event.which) {
+      case 13:
+         input.innerHTML += '\r\n';
+         break;
+      case 9:
+         input.innerHTML += '    ';
+         event.preventDefault();
+         break;
+      case 32:
+         input.innerHTML += ' ';
+         break;
+      case 20:
+         toActiveCapsOrShift(capsLk);
+         event.preventDefault();
+         break;
+      case 8:
+      case 46:
```

Comment:
what happens in cases 8 and 46?

Reply:
Cases 8, 46 and 91 are grupped, because they have the same action - not to add elements to input, because if I didn't write them, the option default would be executed. I wrote this code according [https://learn.javascript.ru/switch#gruppirovka-case](url)